File: app/src/main/java/com/joemerhej/platform/sharedpreferences/SharedPreferencesManager.java
package com.joemerhej.platform.sharedpreferences;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by Joe Merhej on 9/28/18.
 */
public class SharedPreferencesManager
{
    // instance of shared preferences
    private static SharedPreferences mSharedPref;

    // private constants
    private static final String SCHEDULER_PREFIX = "scheduler_";


    private SharedPreferencesManager()
    {
    }

    public static void init(Context context)
    {
        if(mSharedPref == null)
            mSharedPref = context.getApplicationContext().getSharedPreferences(context.getApplicationContext().getPackageName(), Context.MODE_PRIVATE);
    }

    public static String readString(SharedPreferencesKey key, String defaultValue)
    {
        return mSharedPref.getString(SCHEDULER_PREFIX + key.getValue(), defaultValue);
    }

    public static void writeString(SharedPreferencesKey key, String value)
    {
        SharedPreferences.Editor prefsEditor = mSharedPref.edit();
        prefsEditor.putString(SCHEDULER_PREFIX + key.getValue(), value);
        prefsEditor.apply();
    }

    public static boolean readBoolean(SharedPreferencesKey key, boolean defaultValue)
    {
        return mSharedPref.getBoolean(SCHEDULER_PREFIX + key.getValue(), defaultValue);
    }

    public static void writeBoolean(SharedPreferencesKey key, boolean value)
    {
        SharedPreferences.Editor prefsEditor = mSharedPref.edit();
        prefsEditor.putBoolean(SCHEDULER_PREFIX + key.getValue(), value);
        prefsEditor.apply();
    }

    public static int readInt(SharedPreferencesKey key, int defaultValue)
    {
        return mSharedPref.getInt(SCHEDULER_PREFIX + key.getValue(), defaultValue);
    }

    public static void writeInt(SharedPreferencesKey key, int value)
    {
        SharedPreferences.Editor prefsEditor = mSharedPref.edit();
        prefsEditor.putInt(SCHEDULER_PREFIX + key.getValue(), value).apply();
    }

    public static Long readLong(SharedPreferencesKey key)
    {
        String longString = mSharedPref.getString(SCHEDULER_PREFIX + key.getValue(), "");

        Long result = 0L;
        try
        {
            result = Long.valueOf(longString);
        }
        catch(NumberFormatException e)
        {
            return 0L;
        }

        return result;
    }

    public static void writeLong(SharedPreferencesKey key, Long value)
    {
        SharedPreferences.Editor prefsEditor = mSharedPref.edit();
        prefsEditor.putString(SCHEDULER_PREFIX + key.getValue(), value.toString());
        prefsEditor.apply();
    }

    public static void removePreference(SharedPreferencesKey key)
    {
        SharedPreferences.Editor prefsEditor = mSharedPref.edit();
        prefsEditor.remove(SCHEDULER_PREFIX + key.getValue()).apply();
    }

    public static void clearSharedPreferences()
    {
        mSharedPref.edit().clear().apply();
    }
    
}

File: app/src/main/java/com/joemerhej/platform/sharedpreferences/SharedPreferencesManager.kt
package com.joemerhej.platform.sharedpreferences

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by Joe Merhej on 9/28/18.
 */
object SharedPreferencesManager
{
    // instance of shared preferences
    private lateinit var instance: SharedPreferences

    // private constants
    private val SCHEDULER_PREFIX = "scheduler_"


    fun initialize(context: Context)
    {
        instance = context.applicationContext.getSharedPreferences(context.applicationContext.packageName, Context.MODE_PRIVATE)
    }

    fun readString(key: SharedPreferencesKey, defaultValue: String): String?
    {
        return instance.getString(SCHEDULER_PREFIX + key.value, defaultValue)
    }

    fun writeString(key: SharedPreferencesKey, value: String)
    {
        val prefsEditor = instance.edit()
        with(prefsEditor)
        {
            putString(SCHEDULER_PREFIX + key.value, value)
            apply()
        }
    }

    fun readBoolean(key: SharedPreferencesKey, defaultValue: Boolean): Boolean
    {
        return instance.getBoolean(SCHEDULER_PREFIX + key.value, defaultValue)
    }

    fun writeBoolean(key: SharedPreferencesKey, value: Boolean)
    {
        val prefsEditor = instance.edit()
        with(prefsEditor)
        {
            putBoolean(SCHEDULER_PREFIX + key.value, value)
            apply()
        }
    }

    fun readInt(key: SharedPreferencesKey, defaultValue: Int): Int
    {
        return instance.getInt(SCHEDULER_PREFIX + key.value, defaultValue)
    }

    fun writeInt(key: SharedPreferencesKey, value: Int)
    {
        val prefsEditor = instance.edit()
        prefsEditor.putInt(SCHEDULER_PREFIX + key.value, value).apply()
    }

    fun readLong(key: SharedPreferencesKey, defaultValue: Long): Long
    {
        return instance.getLong(SCHEDULER_PREFIX + key.value, defaultValue)
    }

    fun writeLong(key: SharedPreferencesKey, value: Long)
    {
        val prefsEditor = instance.edit()
        prefsEditor.putLong(SCHEDULER_PREFIX + key.value, value).apply()
    }

    fun removePreference(key: SharedPreferencesKey)
    {
        val prefsEditor = instance.edit()
        prefsEditor.remove(SCHEDULER_PREFIX + key.value).apply()
    }

    fun clearSharedPreferences()
    {
        instance.edit().clear().apply()
    }

}
